### PR TITLE
Return cut coords as a list of floats

### PR DESCRIPTION
**Pull request by bcipolli**
`find_xyz_cut_coords` says it returns 3 floats, but it was actually returning 3 one-element arrays.  Tweak the code to return a list of floats.

This necessitated a call to `np.asscalar` in https://github.com/nilearn/nilearn/pull/441 (since changed).

I also added a bit of linting / line breaks while I was in the code.

### CODE REVIEW

File: nilearn/plotting/find_cuts.py [hunk 1]
```diff
@@ -79,22 +82,21 @@ def find_xyz_cut_coords(img, mask=None, activation_threshold=None):
...
+    second_mask = (np.abs(my_map) > second_threshold)
     if second_mask.sum() > 50:
         my_map *= largest_connected_component(second_mask)
     cut_coords = ndimage.center_of_mass(np.abs(my_map))
     x_map, y_map, z_map = cut_coords + offset
 
-    return coord_transform(x_map, y_map, z_map,
-                           img.get_affine())
+    # Return as a list of scalars
+    return np.asarray(coord_transform(x_map, y_map, z_map,
```
Comment (bcipolli) [inline nilearn/plotting/find_cuts.py]: The only functional change is here.

**Comment by lesteve** [comment]
> This necessitated a call to np.asscalar in #441 (since changed).

I realised that the np.asscalar wasn't needed at all in my case. Do you have a use case where it matters whether you have a list of numpy scalar arrays vs a list of floats?

Note that you actually return an array of shape (3, ) not a list. To be honest I am not sure whether there is any guarantee about the return type when the docstring says something like this:

```
Returns
-------
x : float
    the x world coordinate.
y : float
    the y world coordinate.
z : float
    the z world coordinate.
```

**Comment by bcipolli** [comment]
That was the use-case :)  Anotehr is checking to see whether a set of coordinates exists in a list (of lists).  I spent a good chunk of time today debugging matplotlib because of exactly that case--having the numpy object, rather than the actual scalar, broke the `in` check.

So when I saw this just now, it raised red flags.  You can get weird errors.

Also note that `tolist()` is called there, so it should be returning a list (and unpack properly).

**Comment by lesteve** [comment]
> having the numpy object, rather than the actual scalar, broke the in check.

Somehow it seems to work for me, not an expert of numpy scalar arrays though:
```
In [16]: l = [np.array(0), 1]

In [17]: 0 in l
Out[17]: True

In [18]: np.array(1) in l
Out[18]: True
```

> Also note that tolist() is called there, so it should be returning a list (and unpack properly).

I did miss that thanks for pointing it out.

**Comment by bcipolli** [comment]
Cool.  I looked more closely, and apparently the issue wan't quite what I thought it was--only the final case causes a problem, which is not the case here.

```python
>>> (1,2,3) in ((1,2,3),(4,5,6))
True
>>> (np.array(1),2,3) in ((1,2,3),(4,5,6))
True
>>> (np.array(2),2,np.array(3)) in ((1,2,3),(4,5,6))
False
>>> (np.array([1,2,3])) in ((1,2,3),(4,5,6))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

So no--no specific use-case.  Feel free to close (or to cherry-pick the linting)!

**Comment by GaelVaroquaux** [comment]
LGTM! :+1: for merge.

**Comment by AlexandreAbraham** [comment]
Yeah, I would have been puzzled too if I received a list of scalar arrays. Thanks!

**Comment by AlexandreAbraham** [comment]
Sorry, closed instead of merge :P

**Pull request merged**